common/components/AddressBookTableRow.tsx

```diff
@@ -0,0 +1,149 @@
+    this.setState({ label: nextProps.label, mostRecentValidLabel: nextProps.label });
+  }
+
+  public render() {
+    const {
+      address,
+      temporaryLabel,
+      labelError,
+      isEditing,
+      onEditClick,
+      onRemoveClick
+    } = this.props;
+    const { labelInputTouched } = this.state;
+    const trOnClick = isEditing ? noop : onEditClick;
+    const labelInputClassName = labelInputTouched && labelError ? 'invalid' : '';
+    const hashName = `${address}-hash`;
+    const labelName = `${address}-label`;
+
+    return (
+      <React.Fragment>
+        <div className="AddressBookTable-row" onClick={trOnClick}>
+          <div className="AddressBookTable-row-input">
+            <div className="AddressBookTable-row-input-wrapper">
+              <label htmlFor={hashName} className="AddressBookTable-row-input-wrapper-label">
+                {translate('ADDRESS')}
+              </label>
+              <Input name={hashName} title={address} value={address} readOnly={true} />
+            </div>
+            <div className="AddressBookTable-row-identicon AddressBookTable-row-identicon-non-mobile">
+              <Identicon address={address} />
+            </div>
+            <div className="AddressBookTable-row-identicon AddressBookTable-row-identicon-mobile">
+              <Identicon address={address} size="3rem" />
+            </div>
+          </div>
+          <div className="AddressBookTable-row-input">
+            <div className="AddressBookTable-row-input-wrapper">
+              <label htmlFor={labelName} className="AddressBookTable-row-input-wrapper-label">
+                {translate('LABEL')}
+              </label>
+              <Input
+                name={labelName}
+                title={`${translateRaw('EDIT_LABEL_FOR')}${address}`}
+                className={labelInputClassName}
+                value={temporaryLabel}
+                onChange={this.handleLabelChange}
+                onKeyDown={this.handleKeyDown}
+                onFocus={this.setLabelTouched}
+                onBlur={this.handleBlur}
+                showInvalidBeforeBlur={true}
+                setInnerRef={this.setLabelInputRef}
+              />
+            </div>
+            <button
+              title={translateRaw('REMOVE_LABEL')}
+              className="btn btn-sm btn-danger"
+              onClick={onRemoveClick}
+            >
+              <i className="fa fa-close" />
+            </button>
+          </div>
+          {labelError && (
+            <div className="AddressBookTable-row AddressBookTable-row-error AddressBookTable-row-error--mobile">
+              <label className="AddressBookTable-row-input-wrapper-error">
+                <div />
+                {labelError}
+              </label>
+            </div>
+          )}
+        </div>
+        {labelError && (
+          <div className="AddressBookTable-row AddressBookTable-row-error AddressBookTable-row-error--non-mobile">
+            <div />
```

Comment:
What're these empty divs for? If you're trying to make something display: block, better to do that in CSS.

Reply:
:+1: